gcloud/datastore/client.py

```diff
@@ -51,53 +130,168 @@ def __init__(self, dataset_id=None, namespace=None, connection=None):
+        """Retrieve entities, along with their attributes.
 
-        Passes our ``dataset_id``.
+        :type keys: list of :class:`gcloud.datastore.key.Key`
+        :param keys: The keys to be retrieved from the datastore.
+
+        :type missing: an empty list or None.
+        :param missing: If a list is passed, the key-only entities returned
+                        by the backend as "missing" will be copied into it.
+                        Use only as a keyword param.
+
+        :type deferred: an empty list or None.
+        :param deferred: If a list is passed, the keys returned
+                        by the backend as "deferred" will be copied into it.
+                        Use only as a keyword param.
         """
-        return get_multi(keys, missing=missing, deferred=deferred,
-                         connection=self.connection,
-                         dataset_id=self.dataset_id)
+        if not keys:
+            return []
+
+        ids = list(set([key.dataset_id for key in keys]))
+        if ids != [self.dataset_id]:
+            raise ValueError('Keys do not match dataset ID')
+
+        transaction = Transaction.current()
+
+        entity_pbs = _extended_lookup(
+            connection=self.connection,
+            dataset_id=self.dataset_id,
+            key_pbs=[k.to_protobuf() for k in keys],
+            missing=missing,
+            deferred=deferred,
+            transaction_id=transaction and transaction.id,
+        )
+
+        if missing is not None:
+            missing[:] = [
+                helpers.entity_from_protobuf(missed_pb)
+                for missed_pb in missing]
+
+        if deferred is not None:
+            deferred[:] = [
+                helpers.key_from_protobuf(deferred_pb)
+                for deferred_pb in deferred]
+
+        return [helpers.entity_from_protobuf(entity_pb)
+                for entity_pb in entity_pbs]
 
     def put(self, entity):
-        """Proxy to :func:`gcloud.datastore.api.put`.
+        """Save an entity in the Cloud Datastore.
 
-        Passes our ``dataset_id``.
+        .. note::
+
+           This is just a thin wrapper over :meth:`put_multi`.
+           The backend API does not make a distinction between a single
+           entity or multiple entities in a commit request.
+
+        :type entity: :class:`gcloud.datastore.entity.Entity`
+        :param entity: The entity to be saved to the datastore.
         """
-        return put(entity, connection=self.connection,
-                   dataset_id=self.dataset_id)
+        self.put_multi(entities=[entity])
 
     def put_multi(self, entities):
-        """Proxy to :func:`gcloud.datastore.api.put_multi`.
+        """Save entities in the Cloud Datastore.
 
-        Passes our ``dataset_id``.
+        :type entities: list of :class:`gcloud.datastore.entity.Entity`
+        :param entities: The entities to be saved to the datastore.
         """
```

Comment:
In the split view, this is on `put_multi`, but maybe you are seeing something different?